test/dynamic_config_test.go

```diff
@@ -263,3 +516,193 @@ func getPodIDs(ctx context.Context, aeroCluster *asdbv1.AerospikeCluster) (map[s
+
+	return updateCluster(k8sClient, ctx, aeroCluster)
+}
+
+func validateNetworkContextDynamically(
+	ctx goctx.Context, flatServer, flatSpec *asconfig.Conf,
+	aeroCluster *asdbv1.AerospikeCluster, dynamic mapset.Set[string],
+) error {
+	newSpec := *flatSpec
+	ignoredConf := mapset.NewSet("connect-timeout-ms")
+
+	for confKey, val := range *flatServer {
+		if asconfig.ContextKey(confKey) != "network" {
+			continue
+		}
+
+		tokens := strings.Split(confKey, ".")
+
+		if dyn := dynamic.Contains(asconfig.GetFlatKey(tokens)); dyn && !ignoredConf.Contains(asconfig.BaseKey(confKey)) {
+			v := updateValue(val)
+			if v != nil {
+				newSpec[confKey] = v
+			}
+		}
+	}
+
+	newConf := asconfig.New(logger, &newSpec)
+	newMap := *newConf.ToMap()
+
+	aeroCluster.Spec.AerospikeConfig.Value["network"] = lib.DeepCopy(newMap["network"])
+
+	return updateCluster(k8sClient, ctx, aeroCluster)
+}
+
+func validateNamespaceContextDynamically(
+	ctx goctx.Context, flatServer, flatSpec *asconfig.Conf,
+	aeroCluster *asdbv1.AerospikeCluster, dynamic mapset.Set[string],
+) error {
+	newSpec := *flatSpec
+	ignoredConf := mapset.NewSet("rack-id", "default-ttl")
+
+	for confKey, val := range *flatServer {
+		if asconfig.ContextKey(confKey) != "namespaces" {
+			continue
+		}
+
+		tokens := strings.Split(confKey, ".")
+		if dyn := dynamic.Contains(asconfig.GetFlatKey(tokens)); dyn && !ignoredConf.Contains(asconfig.BaseKey(confKey)) {
+			v := updateValue(val)
+			if v != nil {
+				if configWithMaxDefaultVal.Contains(asconfig.BaseKey(confKey)) {
+					v = v.(int64) - 1
+				}
+
+				newSpec[confKey] = v
+			}
+		}
+	}
+
+	newConf := asconfig.New(logger, &newSpec)
+	newMap := *newConf.ToMap()
+
+	aeroCluster.Spec.AerospikeConfig.Value["namespaces"] = lib.DeepCopy(newMap["namespaces"])
+
+	return updateCluster(k8sClient, ctx, aeroCluster)
+}
+
+func validateSecurityContextDynamically(
+	ctx goctx.Context, flatServer, flatSpec *asconfig.Conf,
+	aeroCluster *asdbv1.AerospikeCluster, dynamic mapset.Set[string],
+) error {
+	newSpec := *flatSpec
+
+	for confKey, val := range *flatServer {
+		if asconfig.ContextKey(confKey) != "security" {
+			continue
+		}
+
+		tokens := strings.Split(confKey, ".")
+		if dyn := dynamic.Contains(asconfig.GetFlatKey(tokens)); dyn {
+			v := updateValue(val)
+			if v != nil {
+				newSpec[confKey] = v
+			}
+		}
+	}
+
+	newConf := asconfig.New(logger, &newSpec)
+	newMap := *newConf.ToMap()
+
+	aeroCluster.Spec.AerospikeConfig.Value["security"] = lib.DeepCopy(newMap["security"])
+
+	return updateCluster(k8sClient, ctx, aeroCluster)
+}
+
+func validateXDRContextDynamically(
+	ctx goctx.Context, flatServer, flatSpec *asconfig.Conf,
+	aeroCluster *asdbv1.AerospikeCluster, dynamic mapset.Set[string],
+) error {
+	newSpec := *flatSpec
+
+	for confKey, val := range *flatServer {
+		if asconfig.ContextKey(confKey) != "xdr" {
+			continue
+		}
+
+		tokens := strings.Split(confKey, ".")
+		if dyn := dynamic.Contains(asconfig.GetFlatKey(tokens)); dyn {
```

Comment:
You can simply check `if dynamic.Contains(asconfig.GetFlatKey(tokens)) {}`
You are not using dyn anywhere. Same in other functions.